subsys/caf/modules/sensor_sampler.c

```diff
@@ -101,11 +236,21 @@ static int sample_sensors(int64_t *next_timeout)
 		struct sensor_data *sd = &sensor_data[i];
 		const struct sensor_config *sc = &sensor_configs[i];
 
-		if (sd->sample_timeout <= cur_uptime) {
-			err = sample_sensor(sd->dev, sc);
+		if (atomic_get(&sd->state) == SENSOR_ACTIVE) {
+			if (sd->sample_timeout <= cur_uptime) {
+				err = sample_sensor(sd, sc);
+			} else {
+				k_sem_give(&sample);
```

Comment:
I don't get it... I would expect the call directly in sensor trigger and let the main loop handle everything (sample from the same context).